File: frontend/src/types/department.ts
export interface Location {
    id: string;
    name: string;
}

export interface CreateDepartmentRequest {
    name: string;
    code: string;
    description?: string;
    locationId: string;
}


export interface Department {
    id: string;
    name: string;
    code: string;
    description?: string;
    locationId?: string;
    locationName?: string;
}

File: src/main/java/com/example/starter_project_2025/system/department/entity/Department.java
package com.example.starter_project_2025.system.department.entity;

import com.example.starter_project_2025.system.location.entity.Location;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "departments")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Department {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY) // dùng auto-increment Long
    @Column(nullable = false, updatable = false)
    private Long id;

    @Column(nullable = false, length = 255)
    private String name;

    @Column(nullable = false, length = 50, unique = true)
    private String code;

    @Column(length = 500)
    private String description;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "location_id", nullable = false)
    @JsonIgnoreProperties({"departments", "hibernateLazyInitializer", "handler"})
    private Location location;
}
